packages/service-library/src/servicelib/long_running_tasks/http_endpoint_responses.py

```diff
@@ -25,14 +30,25 @@ async def get_task_result(
     tasks_manager: TasksManager, task_context: TaskContext | None, task_id: TaskId
 ) -> Any:
     try:
-        return tasks_manager.get_task_result(
-            task_id=task_id, with_task_context=task_context
+        task_result = tasks_manager.get_task_result(
+            task_id, with_task_context=task_context
+        )
+        await tasks_manager.remove_task(
+            task_id, with_task_context=task_context, reraise_errors=False
         )
-    finally:
-        # the task is always removed even if an error occurs
+        return task_result
+    except (TaskNotFoundError, TaskNotCompletedError):
+        raise
+    except Exception as exc:
+        # the task raised an exception
+        formatted_traceback = "".join(traceback.format_exception(exc))
+        _logger.info("Task '%s' raised an exception: %s", task_id, formatted_traceback)
+
```

Comment:
why is it logged in between as info? this will also clutter the logs. if it is raised here and whoever uses that client does it correctly then the client shall do the logging not that tool.

Reply:
I tend to not agree with this.

I prefer to be able to troubleshoot the issue once it appears. Since it's an unexpected error I prefer it to be logged closed to the source.

Before this PR everything was swallowed since the error did not travers the HTTP interface any longer (as it did in the past). In a previous PR I've accidentally removed this without thinking about the errors.

The error will be logged here with the full trace. In the client part a warning is displayed. If the exception is not handled also a traceback will appear, which can lead back to the issue in the client.